### PR TITLE
Firecracker: Improve default task sizing

### PR DESCRIPTION
**Pull request by bduffany**
- Account for firecracker VM memory overhead in memory estimate. Experimentally, I found that firecracker needs at least 104 MB of memory to be able to run a trivial task (`pwd`). This PR sets the overhead to 150MB to give us some wiggle room. I think this might depend on the size of `initrd.cpio` + `vmlinux` since those get loaded into memory (?) so in the future we might want to calculate the sizes of those files at build time and derive the estimate from those sizes.
- Increase memory and disk estimates when using docker-in-firecracker. I found that running a `mysql:8.0` image inside a test requires at least 400MB additional memory (on top of the default estimate for a `small` test) and about 4GB additional disk space. I derived the estimate values by multiplying those by 2x and 3x, respectively, since customers may run more images / heavier images in their tests.

---

**Version bump**: Patch 

**Related issues**: https://github.com/buildbuddy-io/buildbuddy-internal/issues/1094

### CODE REVIEW

**Review by tylerwilliams — APPROVED** [review]
LGTM

This seems OK to me, (a little sketch), but I'm ok with it if @vadimberezniker doesn't see any issues.

File: enterprise/server/tasksize/tasksize.go [hunk 1]
```diff
@@ -23,6 +23,12 @@ const (
 	DefaultCPUEstimate      = int64(600)
 	DefaultFreeDiskEstimate = int64(100 * 1e6) // 100 MB
 
+	// Additional resources needed depending on task characteristics
+
+	FirecrackerAdditionalMemEstimateBytes = int64(150 * 1e6) // 150 MB
```
Comment (vadimberezniker) [inline enterprise/server/tasksize/tasksize.go:28]: Can you document what this overhead is?

File: enterprise/server/tasksize/tasksize.go [hunk 2]
```diff
@@ -23,6 +23,12 @@ const (
...
+	DockerAdditionalMemEstimateBytes      = int64(800 * 1e6) // 800 MB
```
Comment (vadimberezniker) [inline enterprise/server/tasksize/tasksize.go:29]: Can you clarify (either via comment or more explicit variable name) that this is the additional memory for running Docker-in-firecracker than anything to do with the the Docker container type.

File: enterprise/server/tasksize/tasksize.go [hunk 3]
```diff
@@ -23,6 +23,12 @@ const (
...
+	DockerAdditionalDiskEstimateBytes     = int64(12 * 1e9)  // 12 GB
```
Comment (vadimberezniker) [inline enterprise/server/tasksize/tasksize.go:30]: We should look into making the default free space we allocate for firecracker larger. Need to check if there are any perf implications. I think we have enough local SSD on the executors that we can afford to be generous with default size as long as there isn't significant performance overhead.

**Review by vadimberezniker — APPROVED** [review]

File: enterprise/server/tasksize/tasksize.go (hunk 2 above)
Comment (bduffany) [reply enterprise/server/tasksize/tasksize.go]: Done

File: enterprise/server/tasksize/tasksize.go (hunk 1 above)
Comment (bduffany) [reply enterprise/server/tasksize/tasksize.go:32]: Done

**Pull request merged**